### PR TITLE
paraview: Add version 5.10.1

### PR DESCRIPTION
**Pull request by herrriehm**
There once had been a bucket for Paraview, but it got removed: e6b9494eba9a88674569601fba3973557e0f5ecc. However, I cannot figure out the reason for this deletion.

This new PR installs correctly. But there are two caveats:

- How can we incorporate the [ParaView Settings Files](https://www.paraview.org/Wiki/ParaView_Settings_Files) into the `persist` directory? There seem to be two files under `%APPDATA%\ParaView`, namely `ParaView-UserSettings.json` and `ParaViewX.Y.ini`. In this filename, `X.Y` are kind of dynamically named. Maybe we could somehow use`$majorVersion.$minorVersion`?
- How can we add `checkver` and `autoupdate`? A machine-readable list of all versions can be found at https://www.paraview.org/files/listing.txt. But I cannot figure out how use this list in `checkver`.

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by issaclin32** [comment]
> There once had been a bucket for Paraview, but it got removed: https://github.com/ScoopInstaller/Extras/commit/e6b9494eba9a88674569601fba3973557e0f5ecc. However, I cannot figure out the reason for this deletion.

I think the author accidentally put it in the PR for *Coppeliasim*, so he deleted it, but he did not send another PR for *Paraview*.

File: bucket/paraview.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        "bin\\pvserver.exe",
+        "bin\\vrpn_server.exe"
+    ],
+    "shortcuts": [
+        [
+            "bin\\paraview.exe",
+            "ParaView"
+        ]
+    ]
+}
```
Comment (issaclin32) [inline bucket/paraview.json:28]: > How can we add checkver and autoupdate? 

```suggestion
    "checkver": {
        "url": "https://www.paraview.org/files/listing.txt",
        "regex": "ParaView-([\\d.]+)-Windows-Python(?<pythonver>[\\d.]+)-msvc(?<msvcver>\\d+)-AMD64\\.zip"
    },
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://www.paraview.org/paraview-downloads/download.php?submit=Download&version=v$majorVersion.$minorVersion&type=binary&os=Windows&downloadFile=ParaView-$version-Windows-Python$matchPythonver-msvc$matchMsvcver-AMD64.zip#/dl.zip",
                "extract_dir": "ParaView-$version-Windows-Python$matchPythonver-msvc$matchMsvcver-AMD64"
            }
        },
        "hash": {
            "url": "https://www.paraview.org/paraview-downloads/download.php?submit=Download&version=v$majorVersion.$minorVersion&type=binary&os=Windows&downloadFile=md5sum.txt",
            "regex": "$md5\\s+$basename"
        }
    }
}
```

**Comment by issaclin32** [comment]
> How can we incorporate the [ParaView Settings Files](https://www.paraview.org/Wiki/ParaView_Settings_Files) into the persist directory?

The config file is under `%appdata%`, we usually don't "persist" this in official buckets, unless it is necessary for the package to work properly (for example, if the app's update machanism conflicts with *Scoop*).

By the way, if you want to persist files like `ParaViewX.Y.ini`, you can copy it in `pre_install` and `pre_uninstall` scripts, such as:
```
"pre_install": "if (Test-Path \"$dir\\ParaView*.ini\") {",
"    ensure \"$persist_dir\"",
"    Copy-Item \"$dir\\ParaView*.ini\" \"$persist_dir\\\",
"}",
"pre_uninstall": "if (Test-Path \"$persist_dir\\ParaView*.ini\") { Copy-Item \"$persist_dir\\ParaView*.ini\" \"$dir\\\" }",
```
`ensure` is a *Scoop* helper function that creates a folder if it does not exist.

File: bucket/paraview.json (hunk 1 above)
Comment (herrriehm) [reply bucket/paraview.json:28]: nice, thank you!

**Comment by herrriehm** [comment]
thank you @issaclin32 for your support!

**Review by issaclin32 — APPROVED** [review]

**Pull request merged**